docs/tools/firmware.sh

```diff
@@ -5,500 +5,789 @@
+					|| (echo && cat <<- EOF && read -r
+						Failed to copy script.
+						Please copy the script manually to the EFI partition using Finder
+						Make sure the name of the script is firmware.sh in the EFI partition
+						Press enter after you have copied"
+						EOF
+						)
 				sudo diskutil unmount "/Volumes/${EFILABEL}/"
-				echo
-				echo -e "Run the following commands or run this script itself in Linux now to set up Wi-Fi :-\n\nsudo mkdir -p /tmp/apple-wifi-efi\nsudo mount /dev/nvme0n1p1 /tmp/apple-wifi-efi\nbash /tmp/apple-wifi-efi/firmware.sh\nsudo umount /tmp/apple-wifi-efi\n"
+				cat <<- EOF
+
+				Run the following commands or run this script itself in Linux now to set up Wi-Fi:
+
+				sudo mkdir -p /tmp/apple-wifi-efi
+				sudo mount /dev/nvme0n1p1 /tmp/apple-wifi-efi
+				bash /tmp/apple-wifi-efi/firmware.sh
+				sudo umount /tmp/apple-wifi-efi
+				EOF
 				;;
 			(2)
 
-				echo -e "\nChecking for missing dependencies"
-				python_check
 				echo -e "\nCreating a tarball of the firmware"
-				python3 "$0" /usr/share/firmware $HOME/Downloads/firmware.tar ${verbose}
-				if [[ (${identifier} = iMac19,1) || (${identifier} = iMac19,2) || (${identifier} = iMacPro1,1) ]]
-		then
-					nvramfile=$(ioreg -l | grep RequestedFiles | cut -d "/" -f 5 | rev | cut -c 4- | rev)
-					txcapblob=$(ioreg -l | grep RequestedFiles | cut -d "/" -f 3 | cut -d "\"" -f 1)
-					cp ${verbose} /usr/share/firmware/wifi/C-4364__s-B2/${nvramfile} "$HOME/Downloads/brcmfmac4364b2-pcie.txt"
-					cp ${verbose} /usr/share/firmware/wifi/C-4364__s-B2/${txcapblob} "$HOME/Downloads/brcmfmac4364b2-pcie.txcap_blob"
-					cd $HOME/Downloads
-					tar --append ${verbose} -f firmware.tar brcmfmac4364b2-pcie.txt
-					tar --append ${verbose} -f firmware.tar brcmfmac4364b2-pcie.txcap_blob
-					rm ${verbose} brcmfmac4364b2-pcie.txt
-					rm ${verbose} brcmfmac4364b2-pcie.txcap_blob
-					cd - >/dev/null
-				fi
-				echo -e "\nFirmware tarball saved to Downloads!"
-				echo -e "\nExtract the tarball contents to /lib/firmware/brcm in Linux and run the following in the Linux terminal:"
-				echo -e "\nsudo modprobe -r brcmfmac_wcc"
-				echo "sudo modprobe -r brcmfmac"
-				echo "sudo modprobe brcmfmac"
-				echo "sudo modprobe -r hci_bcm4377"
-				echo "sudo modprobe hci_bcm4377"
+				create_firmware_archive /usr/share/firmware "$HOME/Downloads/firmware.tar" ${verbose}
+				cat <<- EOF
+
+				Firmware tarball saved to Downloads!
+				Extract the tarball contents to /lib/firmware/brcm in Linux and run the following in the Linux terminal:
+
+				sudo modprobe -r brcmfmac_wcc
+				sudo modprobe -r brcmfmac
+				sudo modprobe brcmfmac
+				sudo modprobe -r hci_bcm4377
+				sudo modprobe hci_bcm4377
+				EOF
 				;;
 			(3)
-				echo -e "\nWhat package manager does your Linux distribution use?"
-				echo -e "\n1. apt"
+				echo -e "\nWhat package manager does your Linux distribution use?\n"
```

Comment:
You probably forgot to use cat <<-EOF here

Reply:
I left it like that on purpose, echo looks more like the final output. I can change it if you want though

Reply:
It's fine that way. I just said cause you were using cat so I thought you missed it by mistake.